### PR TITLE
CSCQVAIN-239: Tap row and alternative actions

### PR DESCRIPTION
**Pull request by jppiiroinen**
Now the tap will open a card which will have the extra functionality.

### CODE REVIEW

**Comment by jppiiroinen** [comment]
This needs some changes

File: src/views/Datasets.vue [hunk 1]
```diff
@@ -47,20 +61,24 @@
...
-				<div class="actions">
-					<b-button variant="primary" size="sm" @click.stop="open(row.item.id)"><font-awesome-icon icon="pen" fixed-width />Edit</b-button>
-					<b-button variant="primary" size="sm" @click.stop="view(row.item.identifier)" :disabled="row.item.identifier == null"><font-awesome-icon icon="external-link-alt" fixed-width />View in Etsin</b-button>
-
-					<b-dropdown variant="primary" right text="More" size="sm">
-						<b-dropdown-item-button size="sm" v-b-modal="'dataset-versions-modal'" @click="activeInModal = row.item.id" :disabled="row.item.versions < 1">
+			<template slot="row-details" slot-scope="row">
+				<b-card>
+					<b-button-group class="row-card-primary-actions">
+						<b-button size="sm" variant="primary" @click.stop="view(row.item.identifier)" :disabled="row.item.identifier == null"><font-awesome-icon icon="external-link-alt" fixed-width />View in Etsin</b-button>
```
Comment (Kauhia) [inline src/views/Datasets.vue]: Is there some hidden character in front of this line?

File: src/views/Datasets.vue [hunk 2]
```diff
@@ -241,8 +256,15 @@ export default {
 				this.isBusy = false
 			}
 		},
-		open(id) { // should maybe later be changed to link so that accessability is better
-			this.$router.push({ name: 'editor', params: { id: id }})
+		editDataset(item) {
+			this.$router.push({ name: 'editor', params: { id: item.id }})
+		},
+		clickedRow(item, fuu) {
```
Comment (Kauhia) [inline src/views/Datasets.vue]: Second parameter

File: src/views/Datasets.vue [hunk 3]
```diff
@@ -241,8 +256,15 @@ export default {
...
+			item._showDetails = !item._showDetails
+		},
+		doNothing(e) {
```
Comment (Kauhia) [inline src/views/Datasets.vue]: Not used anywhere. I think you already found out that there are .stop and .prevent

**Review by Kauhia — CHANGES_REQUESTED** [review]

**Comment by jppiiroinen** [comment]
ready for review

File: src/views/Datasets.vue (hunk 2 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue]: :,D

File: src/views/Datasets.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue]: removing

File: src/views/Datasets.vue (hunk 1 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue]: I will check

File: src/views/Datasets.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue]: should be fixed

File: src/views/Datasets.vue (hunk 2 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue]: should be fixed

File: src/views/Datasets.vue (hunk 1 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue]: should be fixed

File: src/views/Datasets.vue [hunk 4]
```diff
@@ -21,51 +21,111 @@
...
+				ref="datasetTable"
+				:items="datasetList"
+				:fields="fields"
+				striped
+				:tbody-tr-class="rowClass"
+				filter="truthy value"
+				:filter-function="filter"
+				:busy.sync="isBusy"
+				primary-key="id"
+				:tbody-transition-props="{'name': 'datasets-flip'}">
```
Comment (Kauhia) [inline src/views/Datasets.vue:34]: Wasn't this something that Jori said would cause unnecessary render?

File: src/views/Datasets.vue [hunk 5]
```diff
@@ -21,51 +21,111 @@
...
+						<b-button
+							variant="primary"
+							size="sm"
+							@click.stop="editDataset(row.item)">
+							<font-awesome-icon icon="pen" fixed-width />
+							Edit
+						</b-button>
+					</b-button-group>
+					<b-button-group size="sm" class="mr-1">
+						<b-button
```
Comment (Kauhia) [inline src/views/Datasets.vue:96]: Should be link?

**Review by Kauhia — APPROVED** [review]
Few minor comments, nothing to prevent merge.

File: src/views/Datasets.vue (hunk 4 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue:34]: true that is removed in another PR

File: src/views/Datasets.vue (hunk 5 above)
Comment (jppiiroinen) [reply src/views/Datasets.vue:96]: nope

**Pull request merged**